### PR TITLE
issue#3584 added plots for rama_ref and janin_ref

### LINKED ISSUE #3584: add plots for Ramachandran and Janin plots
*Issue*

The example code for [analysis data files](https://docs.mdanalysis.org/2.1.0-dev0/documentation_pages/analysis/data.html?highlight=contourf#analysis-data-files) uses `contourf` but shows no plots. 

*Task*

Add [Ramachandran](https://docs.mdanalysis.org/2.1.0-dev0/documentation_pages/analysis/dihedrals.html#figure-ramachandran) and [Janin](https://docs.mdanalysis.org/2.1.0-dev0/documentation_pages/analysis/dihedrals.html#figure-janin) plot figures.

### PR DESCRIPTION
**Pull request by aditi2906**
Fixes #3584 

Changes made in this Pull Request:
 - Added image for Rama_ref 
 - Added image for Janin_ref
 - Updated figure links in filenames.py

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by aditi2906** [comment]
@[micaela-matta](https://github.com/micaela-matta)
The [code sample ](https://docs.mdanalysis.org/2.1.0-dev0/documentation_pages/analysis/data.html?highlight=contourf#MDAnalysis.analysis.data.filenames.Janin_ref)for Janin_Ref was displaying the following error :"Shapes of x (90, 90) and z (60, 60) do not match"
So I have now accordingly modified the step parameter for X and Y.

File: package/MDAnalysis/analysis/data/filenames.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/data/filenames.py]: Add text that references the newly added figures, such as 
```reST
points. The reference data are shown in :ref:`figure-rama-ref-plot`. An example of analyzed data together with the 
reference data are shown in  :ref:`Ramachandran plot figure <figure-ramachandran>`.
```

File: package/MDAnalysis/analysis/data/filenames.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/data/filenames.py]: add empty line above

Comment (orbeckst) [reply package/MDAnalysis/analysis/data/filenames.py]: replace spaces with dashes in reST reference labels

File: package/MDAnalysis/analysis/data/filenames.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/data/filenames.py]: excellent catch

File: package/MDAnalysis/analysis/data/filenames.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/data/filenames.py]: Same as above: change the text to reference the new figure.

File: package/MDAnalysis/analysis/data/filenames.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/data/filenames.py]: same as above

**Review by orbeckst — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please attend the review comments inline. 

Also add yourself to AUTHORS.

Please check that the resulting docs look correct by looking at the output of the "ReadTheDocs" check where you click on the "Details" link.

File: package/MDAnalysis/analysis/data/filenames.py
Comment (aditi2906) [reply package/MDAnalysis/analysis/data/filenames.py]: Please review the latest code push and let me know if it is resolved.

File: package/MDAnalysis/analysis/data/filenames.py
Comment (aditi2906) [reply package/MDAnalysis/analysis/data/filenames.py]: Please review the latest code push and let me know if it is resolved.

**Comment by aditi2906** [comment]
@pep8speaks I implemented the usage of vertical rulers in my ide now. Please review my latest commit if in case any such violation persists.

**Comment by orbeckst** [comment]
Can you resolve the merge conflicts, please. 

Btw, pep8speaks is an automated bot, not a person. Just read the issues that it finds and fix them until the bot reports no more PEP8 formatting issues.

**Comment by aditi2906** [comment]
@orbeckst I am unable to get though the build_docs check. What would you suggest as the next step to get this PR merged asap?

**Comment by hmacdope** [comment]
@aditi2906  did you try building the docs locally on your computer? Your cross references to the figure are not working. Click on the failed action to view the error. It will be easier to test on your own computer. :)

**Comment by aditi2906** [comment]
Sir, I successfully built the documentation locally and was able to see the changes. Please review and let me know if I need to reference the images in dihedrals.py as well

File: package/MDAnalysis/analysis/data/filenames.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/data/filenames.py]: I looked closely at the Janin plots in https://docs.mdanalysis.org/stable/documentation_pages/analysis/dihedrals.html?highlight=ramachandran#figure-janin and the code https://github.com/MDAnalysis/mdanalysis/blob/9d090e695fae02f40328942648bb06a5e3b6553b/package/MDAnalysis/analysis/dihedrals.py#L663-L667 and realized that we plot the chi angles from 0º to 360º. Could you please adjust the example and your plot accordingly?

File: package/MDAnalysis/analysis/data/filenames.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/data/filenames.py:64]: Please add a short caption to the image. E.g., 
```reST
Reference Ramachandran plot, with contours that contain 90% ("allowed region") 
and 99% ("generously allowed region") of the data points from the reference data set.
```

File: package/MDAnalysis/analysis/data/filenames.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/data/filenames.py:96]: Please add a caption
```reST
Janin reference plot with contours that contain 90% ("allowed region") and 
98% ("generously allowed region") of the data points from the reference data set.
```

**Review by orbeckst — CHANGES_REQUESTED** [review]
Thanks for the changes. I looked closely at the code in [analysis/dihedrals.py](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/analysis/dihedrals.py) and saw that we use custom colors to make the reference regions look nice. The default color bar (as used in the example code) does not work well here. Please
- update the example code in the documentation to use the custom colors 
- redraw the PNG files with the custom colors.

Please also 
- add captions to the figures
- use 0º to 360º for Janin plots (and update the example code)

Thank you!

**Review by orbeckst — APPROVED** [review]
Thanks for addressing my comments. I especially liked that you added degree symbols to the tickmarks and that you increased the size of the images — you demonstrate good attention to detail!

I am approving the PR assuming that CI will all come back green and that a visual check of the build docs looks good. @fiona-naughton or I will make sure that this is the case before merging.

**Comment by orbeckst** [comment]
Visual check of https://mdanalysis--3615.org.readthedocs.build/en/3615/documentation_pages/analysis/data.html looks good and CI is all green.

I'll squash-merge.

**Comment by orbeckst** [comment]
Congratulations on your merged PR, @aditi2906 ! 🏅 👍

**Pull request merged**
